libultraship/libultraship/GlobalCtx2.cpp

```diff
@@ -101,6 +105,17 @@ namespace Ship {
             GetLogger()->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [%@] [%l] %v");
             spdlog::register_logger(GetLogger());
             spdlog::set_default_logger(GetLogger());
+#elif defined(_DEBUG)
```

Comment:
Why separate logging semantics for debug? Seems to me like we should just change the if defined(__linux__) check to defined(__linux__) || defined(_DEBUG)

Reply:
Oh yeah, that's left overs from the early stages of the Wii U port where async logging made debugging crashes more annoying.
On the Wii U port I've disabled file logging to reduce I/O usage to prevent potential slowdowns, and disabled logging entirely for non-debug builds since console prints aren't visible anyways.
Now that the Wii U port features the SohConsole this code could be adjusted.
What do you think of something like this?

```c++
            // Setup Logging
            spdlog::init_thread_pool(8192, 1);
            std::vector<spdlog::sink_ptr> Sinks;

            auto SohConsoleSink = std::make_shared<spdlog::sinks::soh_sink_mt>();
            SohConsoleSink->set_level(spdlog::level::trace);
            Sinks.push_back(SohConsoleSink);

#if !defined(__WIIU__) || defined(_DEBUG)
            auto ConsoleSink = std::make_shared<spdlog::sinks::stdout_color_sink_mt>();
            ConsoleSink->set_level(spdlog::level::trace);
            Sinks.push_back(ConsoleSink);
#endif

#ifndef __WIIU__
            auto logPath = GetPathRelativeToAppDirectory(("logs/" + GetName() + ".log").c_str());
            auto FileSink = std::make_shared<spdlog::sinks::rotating_file_sink_mt>(logPath, 1024 * 1024 * 10, 10);
            FileSink->set_level(spdlog::level::trace);
            Sinks.push_back(FileSink);
#endif

            Logger = std::make_shared<spdlog::async_logger>(GetName(), Sinks.begin(), Sinks.end(), spdlog::thread_pool(), spdlog::async_overflow_policy::block);
            GetLogger()->set_level(spdlog::level::trace);

#ifndef __WIIU__
            GetLogger()->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [%@] [%l] %v");
#else
            GetLogger()->set_pattern("[%s:%#] [%l] %v");
#endif

            spdlog::register_logger(GetLogger());
            spdlog::set_default_logger(GetLogger());
```